From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Rummel <crummel@microsoft.com>
Date: Tue, 12 Jul 2022 14:30:03 -0500
Subject: [PATCH] Use the live FSharp.Core bits to bootstrap - source-build
 doesn't need to match the shipped bootstrap version. See
 https://github.com/dotnet/fsharp/issues/13463 for more details.

---
 src/fsharp/FSharp.Build/FSharp.Build.fsproj | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/fsharp/FSharp.Build/FSharp.Build.fsproj b/src/fsharp/FSharp.Build/FSharp.Build.fsproj
index 7a86fc24b..3026d0511 100644
--- a/src/fsharp/FSharp.Build/FSharp.Build.fsproj
+++ b/src/fsharp/FSharp.Build/FSharp.Build.fsproj
@@ -53,7 +53,7 @@
     <None Include="Microsoft.FSharp.Overrides.NetSdk.targets" CopyToOutputDirectory="PreserveNewest" />
   </ItemGroup>

-  <ItemGroup Condition="'$(Configuration)' == 'Proto'">
+  <ItemGroup Condition="'$(Configuration)' == 'Proto' and '$(DotNetBuildFromSource)' != 'true'">
     <!--
         The FSharp.Build built here may be loaded directly into a shipped Visual Studio, to that end, we cannot
         rely on new API's just being added to FSharp.Core.
@@ -61,7 +61,7 @@
     <PackageReference Include="FSharp.Core" Version="$(FSharpCoreShippedPackageVersionValue)"/>
   </ItemGroup>

-  <ItemGroup  Condition="'$(Configuration)' != 'Proto'">
+  <ItemGroup  Condition="'$(Configuration)' != 'Proto' or '$(DotNetBuildFromSource)' == 'true'">
     <ProjectReference Include="$(MSBuildThisFileDirectory)..\FSharp.Core\FSharp.Core.fsproj" />
   </ItemGroup>


